force_wfmanager/central_pane/analysis_model.py

```diff
@@ -68,31 +68,34 @@ def add_evaluation_step(self, evaluation_step):
 
         self._evaluation_steps.append(evaluation_step)
 
-    def _get_selected_step_index(self):
-        return self._selected_step_index
-
-    def _set_selected_step_index(self, value):
-        if (isinstance(value, int) and not
-                (0 <= value < len(self._evaluation_steps))):
-            raise ValueError(
-                "Invalid value for selection index {}. Current "
-                "number of steps = {}".format(
-                    value,
-                    len(self._evaluation_steps)
-                )
-            )
-
-        self._selected_step_index = value
+    def _get_selected_step_indices(self):
+        return self._selected_step_indices
+
+    def _set_selected_step_indices(self, values):
```

Comment:
Is there any danger in making the assignment before the checks? I spotted a attr (`_plot_index_datasource`) in plot.py which listens to `selected_step_indices`, not sure if that would get updated immediately though.
I guess my thought is that if values contains a 'missing' index it might fail on the chaco side of things?

Reply:
I think you're right, though it's hard to test. I've refactored it so that `selected_step_indices` only gets modified if the `values` passed to `_set_selected_step_indices` are valid.